pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2203,13 +2202,18 @@ public boolean isSingleTenantCluster() {
    * http requests. So, on response, we need mapping from the endpoint to the
    * server instances. With BiMap, both mappings are easily available
    */
-  @Nonnull
-  public BiMap<String, String> getDataInstanceAdminEndpoints(@Nonnull Set<String> instances)
-      throws InvalidConfigException {
+  public @Nonnull
+  BiMap<String, String> getDataInstanceAdminEndpoints(@Nonnull Set<String> instances)
+      throws InstanceNotFoundException, InvalidConfigException {
     Preconditions.checkNotNull(instances);
     BiMap<String, String> endpointToInstance = HashBiMap.create(instances.size());
     for (String instance : instances) {
       InstanceConfig helixInstanceConfig = getHelixInstanceConfig(instance);
+      if (helixInstanceConfig == null) {
```

Comment:
Maybe log an warning and skip this instance?

Reply:
Done